atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+                if not observable_type:
+                    paths.extend(get_observable_paths(observable, f'{current_path}{path}'))
+                else:
+                    paths.append(current_path + path[:-1])
+
+            return paths
+
+        observable_paths = get_observable_paths(observables_group, current_path='')
+        for path in observable_paths:
+            full_path = f'observables.{path}'
+            observable = self._data_parser.get_value(observables_group, path)
+            observable_type = self._data_parser.get_value(observables_group, path).attrs.get('type')
+            observable_name = path.split('.')[0]
+            observable_label = '-'.join(path.split('.')[1:])
+            if observable_type == 'configurational':
+                steps = self._data_parser.get(f'{full_path}.step')
+                if steps is None:
+                    self.logger.warning('Missing step information in some observables. These will not be stored.')
+                    continue
+                thermodynamics_steps = list(set(steps) | set(thermodynamics_steps))
+                times = self._data_parser.get(f'{full_path}.time')
+                values = self._data_parser.get(f'{full_path}.value')
+                if isinstance(values, h5py.Group):
+                    self.logger.warning('Group structures within individual observables not supported. These will not be stored.')
+                    continue
+                for i_step, step in enumerate(steps):
+                    if not self._observable_info[observable_type].get(step):
+                        self._observable_info[observable_type][step] = {}
+                        self._observable_info[observable_type][step]['time'] = times[i_step]
+                    observable_key = f'{observable_name}-{observable_label}' if observable_label else f'{observable_name}'
+                    self._observable_info[observable_type][step][observable_key] = values[i_step]
+            else:
+                if observable_name not in self._observable_info[observable_type].keys():
+                    self._observable_info[observable_type][observable_name] = {}
+                self._observable_info[observable_type][observable_name][observable_label] = {}
+                for key in observable.keys():
+                    observable_attribute = self._data_parser.get(f'{full_path}.{key}')
+                    if isinstance(observable_attribute, h5py.Group):
+                        self.logger.warning('Group structures within individual observables not supported. These will not be stored.')
+                        continue
+                    self._observable_info[observable_type][observable_name][observable_label][key] = observable_attribute
+
+            self.thermodynamics_steps = thermodynamics_steps
+
+    def parse_atomsgroup(self, nomad_sec, h5md_sec_particlesgroup: Group, path_particlesgroup: str):
```

Comment:
nomad_sec type is either System or AtomsGroup.

Reply:
Thanks, done.